check-unused-components.sh

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
```

Comment:
This is great! Nice idea to add this!! 🌟 

Had a couple thoughts, wonder if this should sit in the `/scripts` folder and be added to the package json as a `npm run XXX` type of script maybe? Wdyt?